lib/block.rb

```diff
@@ -0,0 +1,24 @@
+require 'pry'
+require 'date'
+require_relative 'booking'
+
+module Hotel
+  class Block < Booking
+    attr_reader :id, :rooms, :date_range, :nights, :total_cost, :block_info, :block_name, :discount, :reserved_rooms
+
+    def initialize(id, rooms, date_range, block_info)
+      super(id, rooms, date_range)
+      @block_info = block_info
+    end
+
+    def rooms_available_block
+      rooms_available = []
+      @rooms.each do |room|
+        if room.booked == false || room.booked == nil
+          rooms_available << room
+        end
```

Comment:
This method for keeping track of which rooms are reserved is an interesting idea, but I believe you will run into trouble if you have multiple blocks for different dates containing the same rooms. Because the `Room` instances are shared between blocks, marking a room reserved for one block will make it reserved for all blocks.